src/istgt_iscsi.c

```diff
@@ -2049,6 +2053,22 @@ istgt_iscsi_op_login(CONN_Ptr conn, ISCSI_PDU_Ptr pdu)
 			lu = NULL;
 			tsih = 0;
 
+			__sync_add_and_fetch(&discovery_counter, 1);
+
+			/*
+			 * As per the issue https://github.com/openebs/openebs/issues/2382,
+			 * 5 continous discovery requests failure can cause iscsi initiator
+			 * to get into high CPU usage.
+			 * This code checks for this state of 4 continuous discovery failures
+			 * and restarts the process.
+			 */
+			if (discovery_counter > 4) {
+				ISTGT_ERRLOG("discovery counter %lu retry limit exceeded\n",
+				    discovery_counter);
+				sleep(1);
```

Comment:
There is no need to sleep prior to aborting.

Reply:
just taking extra precaution to get the above log synced.

Reply:
Then please call proper functions that enforce this. i.e fflush(3) seems more appropriate to obtain the desired outcome

Reply:
sure.. will do that.. thanks